summer-23-unity/Assets/Scripts/CoinRotation.cs

```diff
@@ -0,0 +1,17 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CoinRotation : MonoBehaviour
+{
+    public float speed = -120.0f;
+    private MeshRenderer model;
+
+    void Start(){
+       model = gameObject.GetComponent<MeshRenderer>();
+    }
+
+    void Update(){
+        model.transform.Rotate(0.0f,0.0f,speed*Time.deltaTime); //rotates only visible mesh, nothing else.
```

Comment:
This seems to rotate all coins even if they have been collected.  is it worrth saving a few cycles to only rotate the coin if its visible/active?

Reply:
The script is attached to each coin individually, and once the coin is destroyed it no longer runs Update